File: Ex-27/ex_27_java/Principal.java
import java.util.Scanner;

public class Principal {

    private Triangulo triang = new Triangulo(new Ponto(0,0),new Ponto(0,0),new Ponto(0,0));

    public void exec() {
        Scanner sc = new Scanner(System.in);
        float x = 0, y = 0;

        System.out.println("Criando um triângulo...");
        System.out.print("Digite a coordenada do primeiro vértice no formato (x.x,y.y): ");
        String entrada = sc.nextLine();
        float[] v1 = parseCoordenada(entrada);
        triang.setVertice1(v1[0], v1[1]);

        System.out.print("Digite a coordenada do segundo vértice no formato (x.x,y.y): ");
        entrada = sc.nextLine();
        float[] v2 = parseCoordenada(entrada);
        triang.setVertice2(v2[0], v2[1]);

        System.out.print("Digite a coordenada do terceiro vértice no formato (x.x,y.y): ");
        entrada = sc.nextLine();
        float[] v3 = parseCoordenada(entrada);
        triang.setVertice3(v3[0], v3[1]);

        triang.imprimirPerimetro();
    }

    // Função auxiliar para converter "(x,y)" em float[]
    private float[] parseCoordenada(String s) {
        s = s.replace("(", "").replace(")", "");
        String[] partes = s.split(",");
        float x = Float.parseFloat(partes[0].trim());
        float y = Float.parseFloat(partes[1].trim());
        return new float[]{x, y};
    }

    public static void main(String[] args) {
        new Principal().exec();
    }
}

File: Ex-27/ex_27_python/Principal.py
from Triangulo import Triangulo


class Principal:
    def __init__(self):
        self.triang = Triangulo()
        self.exec()

    def __del__(self):
        pass

    def exec(self):
        x, y = 0, 0

        print("Criando um triangulo... \nDigite a coordenada do primeiro vertice no formato (x.x,y.y): ", end="")
        x, y = self.lerCoordenada()
        self.triang.setVertice1(x, y)

        print("Digite a coordenada do segundo vertice no formato (x.x,y.y): ", end="")
        x, y = self.lerCoordenada()
        self.triang.setVertice2(x, y)

        print("Digite a coordenada do terceiro vertice no formato (x.x,y.y): ", end="")
        x, y = self.lerCoordenada()
        self.triang.setVertice3(x, y)

        self.triang.imprimirPerimetro()

    def lerCoordenada(self):
        entrada = input().strip().lstrip("(").rstrip(")")
        partes = entrada.split(",")
        return float(partes[0]), float(partes[1])
